### ISSUE TITLE
Add an overload to Threshold.TryParse method

### ISSUE THREAD

**Issue by corrieriluca** [AndreyAkinshin/perfolizer#6]
Hi,

I'm facing an issue with the [`Perfolizer.Mathematics.Thresholds.Threshold.TryParse`](https://github.com/AndreyAkinshin/perfolizer/blob/master/src/Perfolizer/Perfolizer/Mathematics/Thresholds/Threshold.cs#L28) method. I would like to set a custom `CultureInfo` for the string to be parsed as a `Threshold` but I cannot.

Usually `TryParse` methods have multiple overloads (example: [Double.TryParse Method](https://docs.microsoft.com/en-us/dotnet/api/system.double.tryparse?view=net-5.0)) for setting up settings like `NumberStyles` and `IFormatProvider` (which can be a `CultureInfo` for example).

I would like to suggest a change to the `Threshold` API (see below) by adding a overload for the `TryParse` method in order to let the user set these settings. The signature of the "old" `TryParse` method remains and just calls its new overload with default values which has the same body as before but with a call to `double.TryParse` with the new parameters `numberStyle` and `formatProvider`.

```csharp
// This signature still remains and calls the overload
public static bool TryParse(string input, out Threshold parsed)
{
    return TryParse(input, NumberStyles.Any, Thread.CurrentThread.CurrentCulture, out parsed);
}

// New signature
public static bool TryParse(string input, NumberStyles numberStyle, IFormatProvider formatProvider, out Threshold parsed)
{
    if (string.IsNullOrWhiteSpace(input))
    {
        parsed = default;
        return false;
    }

    var trimmed = input.Trim().ToLowerInvariant();
    var number = new string(trimmed.TakeWhile(c => char.IsDigit(c) || c == '.' || c == ',').ToArray());
    var unit = new string(trimmed.SkipWhile(c => char.IsDigit(c) || c == '.' || c == ',' || char.IsWhiteSpace(c)).ToArray());

    if (!double.TryParse(number, numberStyle, formatProvider, out var parsedValue)
        || !ThresholdUnitExtensions.ShortNameToUnit.TryGetValue(unit, out var parsedUnit))
    {
        parsed = default;
        return false;
    }

    parsed = parsedUnit == ThresholdUnit.Ratio ? Create(ThresholdUnit.Ratio, parsedValue / 100.0) : Create(parsedUnit, parsedValue);

    return true;
}
```

What do you think of this addition ?

Is there other locations in the library where a similar change should also be performed ?

I'd be pleased to open a PR to perform these changes, if any.

Best regards.

**Comment by AndreyAkinshin** [comment]
Hi @corrieriluca!

> What do you think of this addition ?

Looks good to me, thanks for the proposal.

> Is there other locations in the library where a similar change should also be performed ?

I guess, it makes sense to introduce similar overloads for `Frequency.TryParse*` methods.

> I'd be pleased to open a PR to perform these changes, if any.

It would be great!

**Comment by corrieriluca** [comment]
Resolved with #7
